app/uk/gov/hmrc/gform/views/form/form.scala.html

```diff
@@ -124,25 +126,32 @@
       }
     }
 
-    @formWithoutCSRF(action = page.formAction, args = 'class -> "js-form", 'autocomplete -> "on", 'novalidate -> "novalidate", 'id -> "gf-form") {
 
     @* this is here for the sole purpose of making it the default submit button, when fields of group exist *@
-    @displaySaveContinueButton("defaultsink", Map("tabindex" -> "-1", "aria-hidden" -> "true"))
+    @displaySaveContinueButton("defaultsink", Map("tabindex" -> "-1", "aria-hidden" -> "true", "form" -> "gf-form"))
 
-      @{page.snippets}
+    @if(page.upscanData.isDefined){
+        @formWithoutCSRF(action = Call("POST", page.upscanData.get.url), args = 'class -> "js-form", 'autocomplete -> "on", 'novalidate -> "novalidate", 'id -> "gf-upscan"){
+            @{page.upscanData.get.snippets}
+        }
+    }
+    @formWithCSRF(action = page.formAction, args = 'class -> "js-form", 'autocomplete -> "on", 'novalidate -> "novalidate", 'id -> "gf-form") {
+        @{page.snippets}
+    }
 
-      <div class="govuk-button-group">
-          @displaySaveContinueButton()
-          @if(page.renderComeBackLater) {
+    <div class="govuk-button-group">
+        @displaySaveContinueButton(otherAttributes = Map("form" -> "gf-form"))
```

Comment:
Can we undo this change and move it back to `gf-form` form?

Reply:
Done